use undoStack as an Array of ChnageSets

denzo opened the pull request.

offirgolan commented:
@denzo there are a lot of changes here. Can you explain what's going on and why these changes are needed?

denzo commented:
@offirgolan Apologies for the delayed response.

I've updated the description of this PR and added tests and explanation in the code.

I hope it made it a bit more clear. Let me know what you think and if this is something you would consider including as a feature of this awesome addon.
